core/src/main/java/smile/classification/RBFNetwork.java

```diff
@@ -156,7 +156,7 @@ public Trainer(Metric<T> distance) {
          * @param rbf the radial basis function.
          * @param m the number of basis functions.
          */
-        public Trainer setRBF(RadialBasisFunction rbf, int m) {
+        public Trainer<T> setRBF(RadialBasisFunction rbf, int m) {
```

Comment:
Sounds like a reasonable default, but I never heard of that, in fact you could also write something like this `public <X> Trainer<X> setRBF(...)` which could potentially make sense if you created a new trainer instance of different type there, so I think even if omitting the generic type parameter is valid you should add it such that this does not lead to problems in future development.

I also just tried that:

```
public class Test<T> {
        T tmp;

        public Test setTmp(final T x) {
            tmp = x;
            return this;
        }
        
        public T getTmp() {
            return tmp;
        }

        public Test<T> setTmp2(T y) {
            tmp = y;
            return this;
        }
    }
```

Now it is perfectly fine to write something like `String str = new Test<String>().setTmp2("abc").getTmp();`
but writing `String str = new Test<String>().setTmp("abc").getTmp();` does only work when you add a cast to `String`